src/jltypes.c

```diff
@@ -756,7 +756,7 @@ static jl_value_t *intersect_typevar(jl_tvar_t *a, jl_value_t *b,
             return (jl_value_t*)a;
         }
         else {
-            jl_tvar_t *new_b = jl_new_typevar(underscore_sym, jl_bottom_type, b);
+            jl_tvar_t *new_b = jl_new_typevar(underscore_sym, jl_bottom_type, b);  // ditto here
```

Comment:
Curiously, after rooting `new_b` I got a [travis segfault](https://travis-ci.org/JuliaLang/julia/jobs/51690923) triggered from this line. Any insights about why? I can remove the root, of course, but my concern is that it's symptomatic of a different problem.